src/main/com/yetanalytics/re_route/listeners.cljs

```diff
@@ -0,0 +1,143 @@
+(ns com.yetanalytics.re-route.listeners
+  (:require [re-frame.core :as re-frame]
+            [com.yetanalytics.re-route.navigation :as nav]
+            [com.yetanalytics.re-route.path :as path]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Event and element
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn- prevent-default! [event]
+  (.preventDefault event))
+
+(defn- set-event-return! [event v]
+  (set! (.-returnValue event) v))
+
+(defn event-target
+  "Get the element associated with the event."
+  ^{:see-also ["reitit.frontend.history/event-target"]}
+  [event]
+  (if (exists? (.-composedPath event))
+    ;; Bottom-most composed path element
+    (aget (.composedPath event) 0)
+    ;; No composed path, just return the event target
+    (.-target event)))
+
+;; `nodeName` is upper case for HTML always,
+;; for XML or XHTML it would be in the original case
+(def anchor-tag "A")
+
+(defn- anchor-element
+  "Return the element (either `element` itself or a parent)
+   with the anchor tag `<a>`."
+  ^{:see-also ["reitit.frontend.history/closest-by-tag"]}
+  [element]
+  (loop [el element]
+    (when el
+      (if (= anchor-tag (.-nodeName el))
+        el
+        (recur (.-parentNode el))))))
+
+(defn ignore-anchor-click?
+  "Return whether the click event's default behavior (to reload the new
+   page) should be ignored or not."
+  ^{:see-also ["reitit.frontend.history/ignore-anchor-click?"]}
+  [event element uri]
+  (and
+   ;; Anchor tag must have href attribute
+   (path/has-scheme? uri)
+   (path/has-domain? uri)
+   ;; Doesn't go to another domain
+   (= (some-> (path/location-uri) path/get-domain)
+      (path/get-domain uri))
+   ;; Not a key + click event
+   (not (.-altKey event))
```

Comment:
this has a smell to it, im not sure if this is the way to go. is there a way to examine the behavior rather than the action?

Reply:
Can you clarify what you mean by this? I merely re-implemented what reitit itself did: https://github.com/metosin/reitit/blob/master/modules/reitit-frontend/src/reitit/frontend/history.cljs#L84

Reply:
i believe all i meant is is there a way to check for ANY modifier universally not iteratively, but if reitit went the same route (hehe) then it's probably reasonably comprehensive for obvious modifiers at least